cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -104,6 +105,15 @@ func GetStaticPodSpecs(cfg *kubeadmapi.MasterConfiguration, k8sVersion *version.
 		}, mounts.GetVolumes(kubeadmconstants.KubeScheduler)),
 	}
 
+	if cfg.CloudProvider == "openstack" {
+		staticPodSpecs[kubeadmconstants.KubeAPIServer].Spec.Containers[0].SecurityContext = &v1.SecurityContext{
+			Privileged: utilpointer.BoolPtr(true),
+		}
+		staticPodSpecs[kubeadmconstants.KubeControllerManager].Spec.Containers[0].SecurityContext = &v1.SecurityContext{
+			Privileged: utilpointer.BoolPtr(true),
```

Comment:
Why does the controller manager need to talk with the meta-data service?

Reply:
when the cloud provider is looking for "which node am i" info, in this specific scenario there is no openstack metadata service and the code looks in the config drive, to be able to mount the config drive it needs privs. we have a `search-order` (https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/) but in this case there is simply no metadata service and the only choice is the config service where it fails because the kube controller manager does not have enough super powers

Reply:
It would be nice to put this down into a code comment.

Reply:
@kargakis ack, if this looks ok otherwise, will happily do that in a little bit

Reply:
Once the cloud-controller code is excised from the controller manager, is there any reason that we would still need this?

Reply:
@timothysc correct, when we excise code that calls out to openstack cloud provider from these 2 containers we can yank the privilege flag.

Reply:
When is the ETA on that excise?  1.10, 1.11?  "Hand-wavy" future?

Reply:
Also, would https://github.com/kubernetes/kubernetes/pull/58080 work for this use case?

Reply:
@timothysc we con't currently expose `--cloud-provider` from kubeadm command line, so that won't help. i've updated the PR with a new field in `MasterConfiguration` and updated the unit test. hopefully that combination will work for everyone.

Reply:
@timothysc For 1.10, i do not expect to move at least the openstack cloud provider out of tree. at best we will have an external openstack controller manager that will work for some scenarios with the in-tree one still being the primary one. 1.11 is a distinct possibility.